lib/jerakia/client.rb

```diff
@@ -93,22 +99,39 @@ def http_send(request)
       end
     end
 
+    def encode_request(request)
+      if not @config.key?(:content_type) or @config[:content_type] == 'json'
+        request.add_field('Content-Type', 'application/json')
+        if request.key?('body')
+          request.body.to_json
+        end
+      elsif @config[:content_type] == 'msgpack'
+        request.add_field('Content-Type', 'application/x-msgpack')
+        if request.key?('body')
+          request.body.to_msgpack
+        end
+      else
+        STDERR.puts("Invalid setting \"#{@config[:content_type]}\" for \":content_type\" in config - supported is either json or msgpack.")
```

Comment:
My experience with Python yesterday has clearly affected me - I meant of course `raise Jerakia::Client::Error, "....."`